Smith_Ben/Assignments/ninja_gold copy/ninjagold.py

```diff
@@ -0,0 +1,44 @@
+from flask import Flask, render_template, session, request, redirect
+import random
+
+app = Flask(__name__)
+app.secret_key = 'gold123'
+
+@app.route('/')
+def index():
+	if not 'gold' in session:
+		session['gold'] = 0
+	if not 'activities' in session:
+		session['activities'] = []
+	return render_template('ninja1.html', )
+
+@app.route('/process', methods = ['POST'])
+def process():
+	locations = {
```

Comment:
Efficiency:  although this all works, the random function is running for each scenario every time which means the CPU is doing 4 times the work that's necessary.  Think about how you would rewrite the code to only calculate gold based on which location is clicked.  You'll have another chance at this in a future stack too.